### DISCUSSION TITLE
How to make pluto work with vegalite interactivity?

### DISCUSSION THREAD

**Discussion by canelhasmateus** [fonsp/Pluto.jl#1521] [Questions about using Pluto] [answered]
Is there any way to make Pluto work with Interactive Vega Scripts? It seems to currently embed a static image. 

I'm talking about configs such as the one below:

<details>
<summary>(click to view example vegalite <code>json</code>)</summary>

```json
{
  "config": {"view": {"continuousWidth": 400, "continuousHeight": 300}},
  "data": {
    "url": "https://cdn.jsdelivr.net/npm/vega-datasets@v1.29.0/data/cars.json"
  },
  "mark": "point",
  "encoding": {
    "color": {
      "condition": {
        "type": "nominal",
        "field": "Origin",
        "selection": "selector064"
      },
      "value": "lightgray"
    },
    "x": {"type": "quantitative", "field": "Miles_per_Gallon"},
    "y": {"type": "quantitative", "field": "Horsepower"}
  },
  "selection": {"selector064": {"type": "interval"}},
  "$schema": "https://vega.github.io/schema/vega-lite/v4.8.1.json"
}
```

</details>

**Comment by eahenle** [comment, marked as answer]
You can generate the `VLSpec` in your notebook and get it in interactive form using `ElectronDisplay`.  Here's your example (plus a title that can be set interactively in the notebook) done two slightly different ways:

![image](https://user-images.githubusercontent.com/16201240/207734838-eddbc77c-a0de-4ef7-9ab1-a30783c31906.png)

In the first approach, I made a `VLSpec` using your exact JSON string, linked the `TITLE` (text box) variable to it with `@set`, and sent it to a separate window, where the plot highlighting works exactly as expected.

In the second approach, I avoided the additional `Setfield` dependency by reformatting the JSON into `VegaLite.jl`'s DSL.  I'm not sure which one I like better; they produce identical outputs.
